### DISCUSSION TITLE
HasManyThroughRelationManager parent record id?

### DISCUSSION THREAD

**Discussion by yaplej** [filamentphp/filament#2635] [Help] [answered]
I am trying to create a HasManyThrough RelationManager but when I create a new record it is missing the parent_id of the Model that owns the relationship.  Should I create a hidden field on my RelationManager Form and populate it somehow with a default value?  I just populated it with a static value to see if it works and it did.

```
    public static function form(Form $form): Form
    {
        return $form
            ->schema([
                Forms\Components\Select::make('node_id')
                    ->required()
                    ->options(Node::all()->pluck('name', 'id'))
                    ->searchable(),
                Forms\Components\Hidden::make('changerequest_id')
                    ->default('1'),
            ]);
    }
```

I tried to look through the Demo but it didnt have any examples of a HasMayThrough relationship.

**Comment by danharrin** [comment, marked as answer]
I don't think you need the parent ID, you just leave that field out completely and Filament will attach the child record to the parent when you save?

**Reply by yaplej** [reply]
I think that the problem is my table structure does not lend itself to HasManyThrough.  In the example the foreign key is stored in the deployments table.
https://laravel.com/docs/9.x/eloquent-relationships#has-many-through

What I am doing is more of a lookup table.  Because deployment could belong to multiple environments using the examples terms.  That must be breaking the Relationship logic a little because it expects a key to exist in the deployments table that I do not have their.

I am going to re-write my Relationship to use a normal HasMany and then use the next relationship off that to pull in the Model I am trying to link.  That should work.
